src/rate-limiter.ts

```diff
@@ -1,17 +1,78 @@
+import { CAPTURE_RATE_LIMIT } from './constants'
+import type { PostHog } from './posthog-core'
 import { RequestResponse } from './types'
 import { logger } from './utils/logger'
 
-const oneMinuteInMilliseconds = 60 * 1000
+const ONE_MINUTE_IN_MILLISECONDS = 60 * 1000
+const RATE_LIMIT_EVENT = '$$js_capture_client_side_rate_limited'
```

Comment:
I checked and events starting with `$$` are still displayed in the event stream. But I can add code in ingestion to translate these into ingestion warnings (as that's the system that should surface this) and drop them so that they're not billed.

If we prefer to not drop them, we could filter out all event names starting with `$$` from the billing query and the app.

Reply:
They aren't billed anyways

Reply:
or actually I cant find the code that ensures they aren't billed... so maybe thats not true anymore 🤔

Reply:
Feels like it would be pretty useful to keep in the event stream tbh. That way you actually have a way of debugging in app (i.e. seeing replay of users with that issue and figuring out whats going wrong)